src/sources/aws_s3/sqs.rs

```diff
@@ -568,24 +568,20 @@ impl IngestorProcess {
             // This handles the transition from the original timestamp logic. Originally the
             // `timestamp_key` was populated by the `last_modified` time on the object, falling
             // back to calling `now()`.
-            match (log_namespace, timestamp) {
-                (LogNamespace::Vector, None) => {
-                    log.insert(metadata_path!("vector", "ingest_timestamp"), Utc::now());
-                }
-                (LogNamespace::Vector, Some(timestamp)) => {
-                    log.insert(metadata_path!(AwsS3Config::NAME, "timestamp"), timestamp);
+            match log_namespace {
+                LogNamespace::Vector => {
+                    if let Some(timestamp) = timestamp {
+                        log.insert(metadata_path!(AwsS3Config::NAME, "timestamp"), timestamp);
+                    }
 
                     log.insert(metadata_path!("vector", "ingest_timestamp"), Utc::now());
                 }
-                (LogNamespace::Legacy, None) => {
+                LogNamespace::Legacy => {
                     log.try_insert(
                         (PathPrefix::Event, log_schema().timestamp_key()),
-                        Utc::now(),
+                        timestamp.unwrap_or_else(Utc::now),
```

Comment:
I only noticed it now (pun intended) , but this isn't a function call anymore (`Utc::now()`) .
I presume the compiler would have caught if that is a mistake but just mentioning it in case.

Reply:
Yeah, `unwrap_or(Utc::now())` triggers a clippy warning.

```text
error: use of `unwrap_or` followed by a function call
   --> src/sources/aws_s3/sqs.rs:580:31
    |
580 |                     timestamp.unwrap_or(Utc::now());
    |                               ^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(Utc::now)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call```

Reply:
`unwrap_or_else` takes a function or closure and calls it. `Utc::now` (no parentheses) is a callable function reference. So this works the same as `|| Utc::now()`.